From de3e1d90a8c7e32df3971b5e0a88909f6d59f37b Mon Sep 17 00:00:00 2001
From: Zhang Peng <peng.zhang1.cn@windriver.com>
Date: Tue, 13 Aug 2024 03:23:07 +0000
Subject: [PATCH 1/2] fix module MyStyle is not installed

```
$ /usr/share/examples/quickcontrols/attachedstyleproperties/attachedstylepropertiesexample
Detected locale "C" with character encoding "ANSI_X3.4-1968", which is not UTF-8.
Qt depends on a UTF-8 locale, and has switched to "C.UTF-8" instead.
If this causes problems, reconfigure your locale. See the locale(1) manual
for more information.
QQmlApplicationEngine failed to load component
qrc:/attachedstyleproperties.qml:7:1: module "MyStyle" is not installed

$ find /usr/share/examples/quickcontrols/attachedstyleproperties
/usr/share/examples/quickcontrols/attachedstyleproperties
/usr/share/examples/quickcontrols/attachedstyleproperties/attachedstylepropertiesexample
/usr/share/examples/quickcontrols/attachedstyleproperties/share
/usr/share/examples/quickcontrols/attachedstyleproperties/share/examples
/usr/share/examples/quickcontrols/attachedstyleproperties/share/examples/quickcontrols
/usr/share/examples/quickcontrols/attachedstyleproperties/share/examples/quickcontrols/attachedstyleproperties
/usr/share/examples/quickcontrols/attachedstyleproperties/share/examples/quickcontrols/attachedstyleproperties/MyStyle
/usr/share/examples/quickcontrols/attachedstyleproperties/share/examples/quickcontrols/attachedstyleproperties/MyStyle/qmldir
/usr/share/examples/quickcontrols/attachedstyleproperties/share/examples/quickcontrols/attachedstyleproperties/MyStyle/libMyStyle.so
/usr/share/examples/quickcontrols/attachedstyleproperties/share/examples/quickcontrols/attachedstyleproperties/MyStyle/libMyStyleplugin.so
```
The module MyStyle incorrectly installed to /usr/share/examples/quickcontrols/attachedstyleproperties/share/examples/
quickcontrols/attachedstyleproperties/MyStyle, tweak INSTALL_EXAMPLEDIR according to the
policy of OE cmake install

```
$ find /usr/share/examples/quickcontrols/attachedstyleproperties
/usr/share/examples/quickcontrols/attachedstyleproperties/attachedstylepropertiesexample
/usr/share/examples/quickcontrols/attachedstyleproperties/MyStyle
/usr/share/examples/quickcontrols/attachedstyleproperties/MyStyle/libMyStyle.so
/usr/share/examples/quickcontrols/attachedstyleproperties/MyStyle/libMyStyleplugin.so
/usr/share/examples/quickcontrols/attachedstyleproperties/MyStyle/qmldir
```

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Zhang Peng <peng.zhang1.cn@windriver.com>
---
 .../attachedstyleproperties/MyStyle/CMakeLists.txt              | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/examples/quickcontrols/attachedstyleproperties/MyStyle/CMakeLists.txt b/examples/quickcontrols/attachedstyleproperties/MyStyle/CMakeLists.txt
index 1448b2d8..922e1941 100644
--- a/examples/quickcontrols/attachedstyleproperties/MyStyle/CMakeLists.txt
+++ b/examples/quickcontrols/attachedstyleproperties/MyStyle/CMakeLists.txt
@@ -2,7 +2,7 @@ if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
     message(FATAL_ERROR "MyStyle should be built as part of the 'attachedstyleproperties' project, and not in isolation.")
 endif()
 
-set(INSTALL_EXAMPLEDIR "${INSTALL_EXAMPLESDIR}/quickcontrols/attachedstyleproperties")
+set(INSTALL_EXAMPLEDIR "${CMAKE_INSTALL_PREFIX}")
 
 qt_policy(SET QTP0001 NEW)
 qt_add_qml_module(MyStyle
-- 
2.44.1

